pkg/kubelet/util/format/pod_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package format
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func fakeCreatePod(name, namespace string, uid types.UID) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			UID:       uid,
+		},
+	}
+}
+
+func fakeCreatePodWithDeletionTimestamp(name, namespace string, uid types.UID, deletionTimestamp *metav1.Time) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              name,
+			Namespace:         namespace,
+			UID:               uid,
+			DeletionTimestamp: deletionTimestamp,
+		},
+	}
+}
+
+func TestPodAndPodDesc(t *testing.T) {
+	testCases := []struct {
+		caseName      string
+		podName       string
+		podNamesapce  string
+		podUID        types.UID
+		expectedValue string
+	}{
+		{"field_empty_but_not_should_be_exist_case", "", "", "", "_()"},
+		{"field_normal_all_case", "test-pod", metav1.NamespaceDefault, "551f5a43-9f2f-11e7-a589-fa163e148d75", "test-pod_default(551f5a43-9f2f-11e7-a589-fa163e148d75)"},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.caseName, func(t *testing.T) {
+			// Create test Ojbect
+			fakePod := fakeCreatePod(testCase.podName, testCase.podNamesapce, testCase.podUID)
+
+			realPod := Pod(fakePod)
+			if realPod != testCase.expectedValue {
+				t.Fatalf("Failed to format pod: Get %s, but want %s", realPod, testCase.expectedValue)
+			}
+
+			realPodDesc := PodDesc(fakePod.Name, fakePod.Namespace, fakePod.UID)
+			if realPodDesc != testCase.expectedValue {
```

Comment:
Please use `assert.Equal` for such comparison (import `github.com/stretchr/testify/assert`)

Reply:
I have tried to change it with 'assert', it look like nice. thank you!